core/src/main/scala/io/odin/syntax/package.scala

```diff
@@ -94,6 +104,15 @@ package object syntax {
     def withContext(implicit clock: Timer[F], monad: Monad[F], withContext: WithContext[F]): Resource[F, Logger[F]] =
       resource.map(ContextualLogger.withContext[F])
 
+    /**
+      * Create conditional logger that buffers messages and sends them to the inner logger when the resource is released.
+      * See [[ConditionalLogger]] for more info
+      */
+    def conditional(
+        fallbackLevel: Level
+    )(implicit timer: Timer[F], F: Concurrent[F], contextShift: ContextShift[F]): Resource[F, Logger[F]] =
+      resource.flatMap(ConditionalLogger.create[F](_, fallbackLevel))
```

Comment:
That's the tricky one.

The outer resource might be not released for the application lifetime. It's totally okay and performance-wise to keep a single instance of `Logger[F]` for an app to save some allocations. Therefore, the conditional logger will never be released as well until the app is killed and no debug messages will be shown. Meanwhile, the unbounded queue will grow _unboundendly_, creating a memory leak.

That's said, I don't think we should provide `.conditional` for `Resource` loggers as it's an easy way to shoot yourself in the leg.

Reply:
Good catch. Better to remove this method.

Reply:
@iRevive what do you think of a syntax like:

```scala
def withBracket(minLevel: Level)(fa: F[A])(implicit F: Bracket[F, Throwable]): F[A]

// and then

logger.withBracket(Level.Debug) {
  //do the dangerously debugging log with the same logic described earlier
}
```

That way we don't need to enforce `Resource` abstraction to the user

Reply:
I'm not sure that this syntax will work. 

The implementation requires a new ConcurrentQueue per underlying execution. That's why I'm using a `Resource#use` closure since I need a new instance of a logger to enqueue messages into a new queue.

Correct me if I'm missing something.

Reply:
So it's essentially the same. The queue is initialized at the beginning of `F[A]` evaluation and then drained in the end. Think of it as `Resource.use` being called by the functions itself instead returned to the user as a value.

Reply:
Possible implementation:

```scala
def withErrorLevel[F[_]: Timer: Concurrent: ContextShift, A](inner: Logger[F], minLevelOnError: Level, maxBufferSize: Option[Int])(fa: F[A]): F[A] = {
  def acquire: F[ConcurrentQueue[F, LoggerMessage]] =
    ConcurrentQueue.withConfig[F, LoggerMessage](queueCapacity, ChannelType.MPSC)

  def use(queue: ConcurrentQueue[F, LoggerMessage]): F[A] =
    fa // ?? how to enqueue messages? inner logger does not know about this queue

  def release(queue: ConcurrentQueue[F, LoggerMessage], exitCase: ExitCase[Throwable]): F[Unit] =
    // drain logic

  Concurrent[F].bracketCase(acquire)(use)(release)
}
```

But the logger does not know anything about underlying queue.

Reply:
Hmm, my bad, you're correct.

It requires to change a signature a bit:
```scala
def withErrorLevel[F[_], A](inner: Logger[F], minLevelOnError: Level, maxBufferSize: Option[Int])(use: Logger[F] => F[A]): F[A]
```

Essentially, the current implementation of the conditional logger is just fine, except we expose `withErrorLevel` API to the user instead of `Resource`-based. `use` parameter might go straight into the `Resource.use`

Reply:
Ah, get it. 
I've pushed changes with a new syntax.